tools/internal_ci/macos/grpc_buildtests_objc.cfg

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Config file for the internal CI (in protobuf text format)
+
+# Location of the continuous shell script in repository.
```

Comment:
ad the "buildtests" in the name - that looks like you're coming up with a completely new category of tests which seems a bit adhoc - please note that the name "basictests" is baked pretty deep into our testing infrastructure in many places (testing scripts, dashboards, etc.) so I'm averse to coming up with yet another name.

I'd like to understand more what are the test categories you're dealing with in ObjC and what's their significance
and then we can try to come up with a solution that's more aligned with how the test infrastructructure is currently setup.

Reply:
From the PR it seems that your currently have 3 categories of tests:
"ios tests", "ios build tests" (which seems to be mostly building examples), and "mac tests" (which are just basictests run on mac?) is that accurate?   Which ones are you planning to run on PRs / on master?

Reply:
According to #18919, there are these 4 categories:
Unit tests
Interop tests
Cronet tests
Mac tests

(and this PR doesn't seem to mention e.g. cronet), so I'm a bit lost here.

Reply:
Currently there are the 4 categories of tests (what you listed above), 3 build test of examples, and one `cfstream_test`. As discussed in previous comment threads above, there will potentially be dbg, opt configs for each of them (and probably also asan and tsan). All of them run on both PRs and master. I did not completely split them according to these categories but kept some of them together, ended up with 3 categories of test "ios tests", "ios build tests" and "mac tests". The reason I pulled ios build tests out from ios tests is to reduce test time. I could merge ios tests and ios build tests in the same "ios tests" category, but that keeps the test time high in the >60 min region.

Reply:
If you want to have "ios tests", "ios build tests" and "mac tests",
I'd prefer these names:

grpc_basictests_objc_ios.cfg
grpc_basictests_objc_mac.cfg
grpc_basictests_objc_examples.cfg

That is assuming you don't want to have dbg and opt tests (which doesn't seem right).
Otherwise you'd need to have _opt and _dbg variant for each (which seems like too many jobs dedicated to objc tests).